### ISSUE TITLE
Navigation block: spacer z-index interferes with keyboard navigation for sub-menu items

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#34097]
### Description

When there is a spacer block below a navigation block, it's z-index being higher causes a confusing user experience when trying to navigation between sub-menu items. cc @tellthemachines 

This was discovered as [part of the ninth call for testing for the FSE Outreach program](https://make.wordpress.org/test/2021/08/11/fse-program-testing-call-9-handling-highered-headers/#comment-1883):

> 0:43:12 :: Nope, it looks like the “spacer” block has a higher z-index and is preventing click events on those above elements.
> 0:44:11 :: Yep — bug: spacer element below sub-menu items messes with z-index?
> Thinking through this more, it makes sense for why that’s happening — the spacer is farther down in the DOM so it will naturally have a higher context. From a UX perspective, this isn’t great though, because I can’t have a spacer below a hovering sub-menu.
> 0:48:25 :: I think that something about this spacer block is affecting keyboard navigation. Maybe? I can’t “move up or down” past certain elements using just the keyboard.

### Step-by-step reproduction instructions

1. Head to the Site Editor.
2. Add a Navigation block.
3. Add a spacer block beneath it. 
4. Begin adding various items to the Navigation block, including a few sub-menu items.
5. Using your keyboard, try to navigate between the sub-menu items and see that the spacer block gets selected at a certain point due to the z-index issue. 

### Screenshots, screen recording, code snippet

You can find a [lengthier video here](https://i.getf.ly/Apu9O1P9) where you can see the issue from around 40 min to the 45 min marker:

https://user-images.githubusercontent.com/26996883/129633876-58455cfc-76de-4544-a5e4-1039d2e13333.mov

### Environment info

- WordPress 5.8.1, Gutenberg 11.2.1, TT1 Blocks
- Chrome
- Desktop

### Pre-checks

- [X] I have searched the existing issues.
- [X] I have tested with all plugins deactivated except Gutenberg.

**Comment by jasmussen** [comment]
Wild one, thanks for the ticket. This one was hard to reproduce, it required the navigation block to be inside a colunms block first — though presumably any nesting container would cause it, as we currently have a `position: relative;` rule applied to any block, which would reset the z-index hierarchy.

Three ways to fix it: 1 — refactor the spacer block to not have a z index. But I recall it being added so that a 1px tall spacer block is still selectable with the cursor. 

2 — refactor the block to not apply position relative. That's added in https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-list/style.scss#L120, and will be a tricky one as I think it's necessary for the block UI. It's also not a guaranteed solution.

3 — tweak the navigation block to handle this better. Turns out #34228 does that. So this PR will be fixed by #34228.

**Comment by jasmussen** [comment]
This issue should be fixed by #34228.

**Comment by annezazu** [comment]
💥 incredible, per usual @jasmussen. Thank you for your quick and thoughtful work.
